lib/request/index.js

```diff
@@ -0,0 +1,105 @@
+const _ = require('lodash'),
+    runtime = require('postman-runtime'),
+    RunSummary = require('../run/summary'),
+    EventEmitter = require('eventemitter3'),
+    CliReporter = require('../reporters/cli'),
+    util = require('./util'),
+
+    /**
+     * This object describes the various events raised by Newman, and what each event argument contains.
+     * Error and cursor are present in all events.
+     *
+     * @type {Object}
+     */
+    runtimeEvents = {
+        start: [],
+        beforeRequest: ['request', 'item'],
+        request: ['response', 'request', 'item', 'cookies', 'history']
+    };
+
+module.exports = function (options, callback) {
+    // validate all options. it is to be noted that `options` parameter is option and is polymorphic
+    (!callback && _.isFunction(options)) && (
+        (callback = options),
+        (options = {})
+    );
+    !_.isFunction(callback) && (callback = _.noop);
+
+    // ensure that the curl command is present before starting a run
+    if (!_.isString(options.curl)) {
```

Comment:
I'm not sure this is what we want... Newman is not just a CLI tool but also a library. If the user has to pass the curl command as a string, then it sets a different expectation than the `run` function on the library where the user can pass options object.

I think we should move the conversion of options to string here instead of the `bin/newman.js`. Benefits:
1. Consistent interface like `run`.
2. Moving the business logic to library instead of in the binary which should only be responsible to add a executable layer on top of the library. The "meat" of the code should lie in the library.

Reply:
Another point to note is that in future we would not need to convert it to string anymore. If we keep this as it is, we will either have to break the API to now send options rather than the string, or support both APIs where we accept string as well as options object. 😬

Reply:
Resolved.